test/github-issues/8409/issue-8409.ts

```diff
@@ -0,0 +1,62 @@
+import { getConnectionManager } from "../../../src";
+import { Connection } from "../../../src/connection/Connection";
+import { closeTestingConnections } from "../../utils/test-utils";
+import { User } from "./entity/User";
+
+describe("github issues > #4753 MySQL Replication Config broken", () => {
+    let connections: Connection[] = [];
+    after(() => closeTestingConnections(connections));
+
+    it("should connect without error when using replication", async () => {
+        const connection = getConnectionManager().create({
+            type: "mysql",
+            ssl: {
+                ca: "GLOBAL",
+                key: "GLOBAL",
+                cert: "GLOBAL",
+            },
+            replication: {
+                master: {
+                    username: "test",
+                    password: "test",
+                    database: "test",
+                    ssl: {
+                        ca: "MASTER",
+                        key: "MASTER",
+                        cert: "MASTER",
+                    },
+                },
+                slaves: [
+                    {
+                        username: "test",
+                        password: "test",
+                        database: "test",
+                        ssl: {
+                            ca: "CHILD",
+                            key: "CHILD",
+                            cert: "CHILD",
+                        },
+                    },
+                    {
+                        username: "test",
+                        password: "test",
+                        database: "test",
+                    },
+                ],
+            },
+            entities: [User],
+        });
+        connections.push(connection);
+        await connection.connect();
+        let result: any = connection;
+        result.driver.poolCluster._nodes.MASTER.pool.config.connectionConfig.ssl.ca.should.be.eql(
```

Comment:
TBH I'm not confident about these kind of test because we are relying on internal mysql stuff that can easily be changed and then we have to find new apis and fix this test.